File: jobintentservicedemo/app/src/main/java/com/example/jobintentservicedemo/MainActivity.java
package com.example.jobintentservicedemo;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    private EditText etInput;
    public static String KEY_INPUT_TEXT = "input_text";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        init();
    }

    @Override
    public void onClick(View view) {
        int viewId = view.getId();

        if (viewId == R.id.btn_enqueue_work) {
            String input = etInput.getText().toString();

            Intent serviceIntent = new Intent(this, MainActivity.class);
            serviceIntent.putExtra(KEY_INPUT_TEXT, input);
            SampleJobIntentService.enqueueWork(this, serviceIntent);
        }
    }

    private void init() {
        etInput = findViewById(R.id.et_input);
        findViewById(R.id.btn_enqueue_work).setOnClickListener(this);
    }
}

File: databasedemo/app/src/main/java/com/example/databasedemo/MainActivity.kt
package com.example.databasedemo

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.example.databasedemo.room.RoomActivity
import com.example.databasedemo.sqlite.SQLiteActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        init()
    }

    private fun init() {
        findViewById<Button>(R.id.btn_sqlite).setOnClickListener {
            startActivity(Intent(this, SQLiteActivity::class.java))
        }

        findViewById<Button>(R.id.btn_room).setOnClickListener {
            startActivity(Intent(this, RoomActivity::class.java))
        }
    }
}